### ISSUE TITLE
Fix S1006 FP: default parameter is not being overridden

### ISSUE THREAD

**Issue by richardissimo** [SonarSource/sonar-dotnet#9123]
### Description

S1006 describes how it would be a mistake for an override to change a parameter's default. It also mentions that it can lead to unexpected behaviour when an interface is explicitly implemented. All good so far. However, this is being flagged where the parameter is not being overridden.

### Repro steps
Example:
```
    internal static class Program
    {
        static void Main(string[] args)
        {
            IMyProcessorInterface p1 = new MyProcessor1();
            p1.Process();

            IMyProcessorInterface p2 = new MyProcessor2();
            p2.Process();
        }
    }

    public interface IMyProcessorInterface
    {
        void Process(string? message = null);
    }

    public class MyProcessor1 : IMyProcessorInterface
    {
        public void Process(string? message)
        {
            Console.WriteLine(message ?? "null");
        }
    }

    public class MyProcessor2 : IMyProcessorInterface
    {
        public void Process(string? message)
        {
            Console.WriteLine(message ?? "null");
        }
    }
```
This demonstrates where I have two implementations of an interface. The interface is declaring the default parameter, and this is not being overridden by the implementations, so this should not lead to a diagnostic.

### Expected behavior

The above code should not cause a diagnostic to be raised. As the code stands, the default value is in a single place. By following the diagnostic's advice, that value is in 3 places (breaking "Don't Repeat Yourself"). And there is no possible confusion here: if I have a specifically-typed variable, the following code does not compile (and I wouldn't want it to, because it's the interface providing the default parameter, not the implementation.
```
            MyProcessor2 p3 = new MyProcessor2();
            p3.Process();
```
And if I then overrode the default value on MyProcessor2's Process method (so that the above code would compile; but at which point it would behave in the potentially confusing way), then I would be in breach of this rule, and at that point, I would expect an S1006 violation (which I do get). But as long as I'm not overriding the default value, there should not be a complaint.

### Actual behavior

S1006 raised on the "message" parameters of MyProcessor1 and MyProcessor2.

### Known workarounds

Suppress complaint.

### Related information

* SonarAnalyzer.CSharp Version=9.23.2.88755
* Visual Studio version 17.9.5
* dotnet 6.0
* SonarScanner for .NET version (if used)  n/a
* Operating System Windows 10
